butane_cli/src/lib.rs

```diff
@@ -259,10 +261,46 @@ pub fn get_migrations() -> Result<FsMigrations> {
     Ok(migrations::from_root(root))
 }
 
+/// The members of a workspace are given a PackageId, which contains the directory
+/// name however it needs to be extracted.
+pub fn extract_package_directory(
+    package_id: cargo_metadata::PackageId,
+) -> Result<std::path::PathBuf> {
+    let mut file_url = package_id
+        .repr
+        .split_once(&"path+".to_string())
+        .unwrap()
+        .1
+        .to_string();
+    file_url.truncate(file_url.len() - 1);
+    let url = Url::parse(file_url.as_str()).expect("Failed to parse URL");
+    let file_path = url.to_file_path().unwrap();
+    Ok(file_path)
+}
+
+/// Find a .butane directory to act as the base for butane.
 pub fn base_dir() -> Result<PathBuf> {
-    std::env::current_dir()
-        .map(|d| d.join(".butane"))
-        .map_err(|e| e.into())
+    let current_directory = std::env::current_dir()?;
+    let local_butane_dir = current_directory.join(".butane/");
+
+    if !local_butane_dir.exists() {
+        let metadata = MetadataCommand::new().no_deps().exec()?;
+        let workspace_members = metadata.workspace_members;
+
+        // Find the first workspace member with a .butane
+        for member in workspace_members {
+            let package_dir = extract_package_directory(member)?;
+            let member_butane_dir = package_dir.join(".butane/");
+
+            if member_butane_dir.exists() {
+                eprintln!("Using workspace member {:?}", member_butane_dir);
```

Comment:
I think this should go to stdout rather than stderr as it's informational, not fatal/important.